src/pages/lesson-page/lesson-page.tsx

```diff
@@ -14,35 +14,38 @@ export const LessonPage: React.FC = () => {
     const { data } = useLessonBySlug(slug);
     const { data: availability, isLoading } = useAvailability(data?._id!);
 
-    const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+    const [selectedDate, setSelectedDate] = useState<Date | null>(null);
     const [selectedHour, setSelectedHour] = useState<string>('');
 
-    const handledDateSelected = (date: Date | undefined) => {
+    const handledDateSelected = (date: Date | null) => {
         if (date) {
-            setSelectedDate(date);
+            if (selectedDate && isSameDay(date, selectedDate)) setSelectedDate(null);
+            else {
+                setSelectedDate(date);
+                setSelectedHour('');
+            }
+        } else {
+            setSelectedDate(null);
             setSelectedHour('');
         }
     };
 
     const typeOfLesson = data?.name!;
 
-    const datesToLessons = availability?.availabilityEntries?.reduce((acc, entry) => {
-        const currentDay = new Date(entry?.slot?.startDate!);
-        const normalizedDay = format(currentDay, 'yyyy-MM-dd');
-        const lessonStartingHour = format(currentDay, 'HH:mm aa');
+    const datesToLessons = useMemo(() => {
+        return availability?.availabilityEntries?.reduce((acc, entry) => {
```

Comment:
you might have a bug, not sure that you will ever have more than one slot per day in the map